services/workflow-manager/app/api/workflow.py

```diff
@@ -141,3 +252,133 @@ async def get_image_file(record_id: int) -> StreamingResponse:
         return response
     except FileNotFoundError as exc:
         raise HTTPException(detail="File not found.", status_code=status.HTTP_404_NOT_FOUND) from exc
+
+## Formerly acquisition control
+
+async def device_location_request(device_id):
+    """Retrieve ip from device-manager.
+
+    Parameters
+    ----------
+    device_id
+        Id of device
+
+    Returns
+    -------
+        ip_address of device
+    """
+    async with httpx.AsyncClient() as client:
+        response = await client.get(f"http://api-gateway:8080/api/v1/device/{device_id}/ip_address")
+        return response.json()["ip_address"]
+
+
+async def retrieve_sequence(sequence_manager_uri, sequence_id):
+    """Retrieve sequence and sequence-type from sequence-manager.
+
+    Parameters
+    ----------
+    sequence_manager_uri
+        uri of sequence manager
+
+    sequence_id
+        id of sequence
+
+    Returns
+    -------
+        sequence
+    """
+    async with httpx.AsyncClient() as client:
+        response = await client.get(f"http://{sequence_manager_uri}/api/v1/mri/sequences/{sequence_id}")
+        return response.json()
+
+
+async def create_record(exam_manager_uri, job_id):
+    """Create new record at exam_manager and retrieve record_id.
+
+    Parameters
+    ----------
+    exam_manager_uri
+        uri of sequence manager
+
+    job_id
+        id of job
+
+    Returns
+    -------
+        id of newly created record
+    """
+    async with httpx.AsyncClient() as client:
+        # TODO: data_path, comment ? # pylint: disable=fixme
+        data = {
+            "data_path": "unknown",
+            "comment": "Created in Acquisition Control",
+            "job_id": str(job_id),
+        }
+        response = await client.post(f"http://{exam_manager_uri}/api/v1/exam/record", json=data)
```

Comment:
Is this endpoint already implemented?

Reply:
Right, that was the old implementation; Johannes will care about this - I will set a dummy for now.